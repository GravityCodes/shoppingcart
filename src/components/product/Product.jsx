import PropTypes from "prop-types";
import styles from "./Product.module.css"
import { Check } from "lucide-react";
import { useState } from "react";



export const Product = (props) => {
  const [addToCart, setAddToCart] = useState(false);

  const buttonClickHandler = () => {
    props.addToCart(props.title, 1, props.image, props.price, props.id);
    if(addToCart == false) {
      setAddToCart(true);
      setTimeout(() => {
        setAddToCart(false)}, 3000);
    } 
  }

  return (
    <div className={`${styles["product-wrapper"]} `} >
        {addToCart && 
          <div className={styles["added-to-cart"]}>
           <Check color="green" size={15}/> Added to cart 
          </div>}
        <div className={styles.img}>
          <img src={props.image} alt="" />
        </div>
        <div className={styles.info}>
          <p className={styles.price}>${props.price}</p>
          <p className={styles.title}>{props.title}</p>
          <p className={styles.category}>{props.category}</p>
          <button className={styles.button} onClick={buttonClickHandler}>Add to Cart</button>
        </div>
    </div>
  )
}


Product.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  addToCart: PropTypes.func.isRequired,
  id: PropTypes.number,
}