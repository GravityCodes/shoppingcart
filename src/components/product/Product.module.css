.product-wrapper {
  display: flex;
  flex-direction: column;
  box-shadow: 0px 2px 5px rgb(138, 138, 138);
  justify-content:flex-end;
  position: relative;
}

.img {
  flex: 0 3 0;
  padding: 10px;
  max-height: 50%;
  margin: auto 0;
}

.img img {
  margin: 0 auto;
  height: 100%;
}

.info {
  padding: 10px;
}

.title {
  font-size: .7rem;
}

.category {
  font-size: .6rem;
}

.button {
  background-color: black;
  color: white;
  cursor: pointer;
  font-size: .7rem;
  padding: 5px;
}

.button:active {
  background-color: white;
  outline: 1px solid black;
  color: black;
}

.added-to-cart {
  position: absolute;
  top: 0px;
  left: 0px;
  background-color: rgb(187, 251, 186);
  display: flex;
  padding: 5px;
  color: green;
  align-content: center;
  font-size: 0.7rem;
  transform: translateX(0px);
  animation: pop-in .1s ease-in, pop-out .2s ease-out 2s;
  animation-fill-mode: forwards;

}

@keyframes pop-in {
  from {
    transform:  translateX(-10px);
    opacity: 0.5;
  }
  to {
    transform: translateX(0px);
    opacity: 1;
  }
}

@keyframes pop-out {
  from {
    transform: translateY(0px);
    opacity: 1;
  }
  to {
    transform: translateY(-10px);
    opacity: 0;
  }
}