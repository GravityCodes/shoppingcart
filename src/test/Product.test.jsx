import { vi,  describe, it, expect } from "vitest"
import {render, screen} from "@testing-library/react"
import { Product } from "../components/product/Product"
import userEvent from "@testing-library/user-event";

describe("product component", () => {
  it("matches snapshot", () => {

    const addToCart = vi.fn();

    const {container} = render(<Product image="" title="" category="" price={1} addToCart={addToCart} id={1}/>);

    expect(container).toMatchSnapshot();
  });

  it("button calls addToCart function"),async () => {
    const addToCart =  vi.fn();

    const  user= userEvent.setup;
    render(<Product addToCart={addToCart}/>);
 
    let button  = screen.getByRole("button");

    await user.click(button);

    


  }
});
