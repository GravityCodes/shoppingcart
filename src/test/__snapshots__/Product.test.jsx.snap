// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`product component > matches snapshot 1`] = `
<div>
  <div
    class="_product-wrapper_229001 "
  >
    <div
      class="_img_229001"
    >
      <img
        alt=""
        src=""
      />
    </div>
    <div
      class="_info_229001"
    >
      <p
        class="_price_229001"
      >
        $
        1
      </p>
      <p
        class="_title_229001"
      />
      <p
        class="_category_229001"
      />
      <button
        class="_button_229001"
      >
        Add to Cart
      </button>
    </div>
  </div>
</div>
`;
